### ISSUE TITLE
modifier un profil => boucle presque infinie

### ISSUE THREAD

**Issue by Fil** [seenthis/seenthis_squelettes#95]
Dans (au moins) deux fonctions appelées lorsqu'on édite son profil (formulaires/profil.php : nettoyer_nom_auteur et nettoyer_logo_auteur), on trouve une boucle sur l'ensemble des messages qu'on a postés, et qui vise à recalculer leur cache. Quand on en a des milliers autant dire que ça ne peut pas aller au bout.
```
	$query = sql_select("id_me", "spip_me", "id_auteur=$id_auteur");
```
il faudrait remplacer ça par un traitement par morceaux, envoyé en job_queue

**Comment by archiloque** [comment]
C'est la suppression du cache qui est lente ?

**Comment by Fil** [comment]
Oui, la suppression de chaque cache est un peu lente (notamment parce qu'elle provoque au moins une requete sql) ; et quand tu en as qqs centaines c'est déjà trop et tout casse. Ca fait deux problèmes, en fait :8ball:

**Comment by Fil** [comment]
on devrait certainement éviter d'avoir à supprimer des caches, surtout par milliers… soit on fait un inclure d'un truc unique, et on tue ce cache-là ; soit on s'arrange pour que le champ `maj` de spip_me et de spip_auteurs soit passé en contexte au cache (et alors on n'a rien à effacer du tout).

**Comment by archiloque** [comment]
je reformule: si toutes les infos sont dans le contexte du coup les caches sont supprimés au fil de l'eau et c'est moins grave ?

**Comment by Fil** [comment]
Oui exactement ; surtout qu'on n'utilise plus le "microcache" qui écrivait
tout sur le disque sans garabage collector, pour passer à un truc qui
utilise des choses comme memcached.

-- Fil

2015-03-16 12:35 GMT+01:00 Julien Kirch <notifications@github.com>:

**Comment by archiloque** [comment]
Tu utilises quoi sur seenthis pour le cache ?

**Comment by Fil** [comment]
microcache appelle memoization qui parle à memcached

-- Fil

**Comment by archiloque** [comment]
On pourrait commencer par optimizer les requêtes SQL et utiliser l'API memcached pour supprimer les éléments par listes, et passer en job ensuite ?

**Comment by Fil** [comment]
je crois que le bug vient surtout de l'utilisation cache_me() qui invalide
mais aussi recalcule chaque cache ; invalider ça peut aller vite, mais
recalculer... ouille. C'est là qu'on fait des requetes en trop

-- Fil

2015-03-16 14:18 GMT+01:00 Julien Kirch <notifications@github.com>:

**Comment by archiloque** [comment]
Donc si on optimize et qu'on enlève le recalcul on est bon ?

**Comment by Fil** [comment]
bah je sais pas :)  faut voir


-- Fil

**Comment by archiloque** [comment]
je propose de commencer par ça car ça sert de toutes façons

**Comment by brunob** [comment]
On en est où avec ce bug ?
